File: Review/Unit 3/MiniProject/PlaylistObjC/PlaylistObjC/DetailPlaylistTableViewController.m
//
//  DetailPlaylistTableViewController.m
//  PlaylistObjC
//
//  Created by Kaleo Kim on 4/21/16.
//  Copyright © 2016 Kaleo Kim. All rights reserved.
//

#import "DetailPlaylistTableViewController.h"

@interface DetailPlaylistTableViewController ()

@property (strong, nonatomic) IBOutlet UITextField *songNameTextField;
@property (strong, nonatomic) IBOutlet UITextField *artistNameTextField;

@end

@implementation DetailPlaylistTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)addSongButtonTapped:(id)sender {
    
    [[PlaylistController sharedInstance] addSongWithTitle:self.songNameTextField.text andArtist:self.artistNameTextField.text toPLaylist:self.playlist];
    [self.tableView reloadData];
    
}

#pragma mark - Table view data source


- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {

    return [self.playlist.songs count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"playlistDetailCell" forIndexPath:indexPath];
    
    Song *song = self.playlist.songs[indexPath.row];
    
    cell.textLabel.text = song.name;
    cell.detailTextLabel.text = song.artistName;
    
    
    return cell;
}


- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return YES;
}

- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        
        [[PlaylistController sharedInstance] seleteSong:self.playlist.songs[indexPath.row] fromPLaylist:self.playlist];
        
        [tableView reloadData];
    }
}

@end

File: Review/Unit 2/Project/PlaylistMiniProject 2/DetailPlaylistTableViewController.swift
//
//  DetailPlaylistTableViewController.swift
//  PlaylistMiniProject
//
//  Created by admin on 4/6/16.
//  Copyright © 2016 Brock. All rights reserved.
//

import UIKit

class DetailPlaylistTableViewController: UITableViewController {
    
    var playlist: Playlist?
    
    var songs: [Song] {
        guard let playlist = playlist else {return []}
        return Array(playlist.song) as? [Song] ?? []
    }
    
    @IBOutlet weak var artistNameTextField: UITextField!
    @IBOutlet weak var songNameTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false

        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func addSongButtonTapped(_ sender: AnyObject) {
        guard let playlist = playlist,
                  let songTitle = songNameTextField.text,
                  let artistName = artistNameTextField.text else {return}
        SongController.createSong(songTitle, artist: artistName, playlist: playlist)
        //SongController.addSongToPlaylist(song, playlist: playlist)
        tableView.reloadData()
        
    }
    

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // #warning Incomplete implementation, return the number of rows
        //guard let playlist = playlist else {return 0}
        return songs.count
    }

    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "playlistDetailCell", for: indexPath)
    
            let song = songs[(indexPath as NSIndexPath).row]
            cell.textLabel?.text = song.title
            cell.detailTextLabel?.text = song.artist
        
        
        return cell
    }
    

}
